build.helpers.psm1

```diff
@@ -0,0 +1,814 @@
+
+    .OUTPUTS
+        None. Invokes MSBuild and optionally signs the package.
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(
+            Mandatory = $true,
+            Position = 0,
+            HelpMessage = "Path to the Visual Studio solution file"
+        )]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript({
+                if (Test-Path $_ -PathType Leaf) { $true }
+                else { throw "Solution file '$_' does not exist." }
+            })]
+        [string]$SolutionPath,
+
+        [Parameter(
+            Mandatory = $true,
+            Position = 1,
+            HelpMessage = "Platform architecture (x64, x86, arm64)"
+        )]
+        [ValidateSet("x64", "x86", "arm64")]
+        [string]$Platform,
+
+        [Parameter(
+            Mandatory = $true,
+            Position = 2,
+            HelpMessage = "Build configuration (Debug, Release)"
+        )]
+        [ValidateSet("Debug", "Release")]
+        [string]$Configuration,
+
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = "Output directory for the MSIX package"
+        )]
+        [ValidateNotNullOrEmpty()]
+        [string]$OutputDirectory,
+
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = "Build ring identifier (Dev, Stable, etc.)"
+        )]
+        [ValidateNotNullOrEmpty()]
+        [string]$BuildRing,
+
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = "Path to MSBuild.exe"
+        )]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript({
+                if (Test-Path $_ -PathType Leaf) { $true }
+                else { throw "MSBuild.exe not found at '$_'." }
+            })]
+        [string]$MSBuildPath
+    )
+
+    begin
+    {
+        Write-Verbose "Starting MSBuild package build"
+        Write-Verbose "Platform: $Platform, Configuration: $Configuration"
+        Write-Verbose "Output Directory: $OutputDirectory"
+        Write-Verbose "Build Ring: $BuildRing"
+    }
+
+    process
+    {
+        if (-not (Test-Path $OutputDirectory))
+        {
+            Write-Verbose "Creating output directory: $OutputDirectory"
+            New-Item -Path $OutputDirectory -ItemType Directory -Force | Out-Null
+        }
+
+        $packagePath = Join-Path $OutputDirectory "WinGetStudio-$Platform.msix"
+        Write-Verbose "Package output path: $packagePath"
+
+        $msbuildArgs = @(
+            $SolutionPath,
+            "/p:Platform=$Platform",
+            "/p:Configuration=$Configuration",
+            "/p:RestorePackages=false"
+            "/binaryLogger:WinGetStudio.$Platform.$Configuration.binlog",
+            "/p:AppxPackageOutput=$packagePath",
+            "/p:AppxPackageSigningEnabled=false",
+            # "/p:GenerateAppxPackageOnBuild=true",
```

Comment:
Can we instead keep this here and remove it from the csproj? I think we only ever need it for the pipeline